docs/spec.md

```diff
@@ -2813,8 +2814,9 @@ More formally, `result[result_index] = operand[operand_index]` where:
     `index_vector_dim` < `rank(start_indices)`.
   * `[start_indices[batch_index]]` otherwise.
 * For `d_operand` in `axes(operand)`,
-  * `full_start_index[d_operand]` = `start_index[d_start]` if
-    `d_operand = start_index_map[d_start]`.
+  * `full_start_index[d_operand]` =
+    `clamp(start_index[d_start], 0, dim(operand, d_operand) - slice_sizes[d_operand])`
+    if `d_operand = start_index_map[d_start]`.
```

Comment:
Also how about we move clamping to a separate bullet, i.e. 

```
* `start_index` =
  * `start_indices[bi0, ..., :, ..., biN]` where `bi` are individual elements in
    `batch_index` and `:` is inserted at the `index_vector_dim` index, if
    `index_vector_dim` < `rank(start_indices)`.
  * `[start_indices[batch_index]]` otherwise.
* `clamped_start_index = ...`.
* For `d_operand` in `axes(operand)`,
  * `full_start_index[d_operand]` = `clamped_start_index[d_start]` if
    `d_operand = start_index_map[d_start]`.
```

This will make the intent more explicit and will also avoid the need for markdownlint-disable line-length.

Reply:
Oops, I see now that clamping cannot happen before applying start_index_map - it should happen afterwards. I still think that modularization could be useful though, along the lines of `clamped_full_start_index = ...`.

Reply:
Okay nevermind - I now think that moving clamping into a separate bullet isn't worth it overall, because it would make updating the associated image considerably harder and the benefits are pretty minor all things considered. Resolving this thread.